frontend/src/pages/RegisterPage.tsx

```diff
@@ -0,0 +1,125 @@
+import { Button } from "@/components/ui/button"
+import useAuth from "@/context/AuthContext"
+import { useState } from "react"
+import { Link } from "react-router-dom"
+
+export type Error = {
+  id?: number
+  message?: string
+}
+
+function RegisterPage() {
+  const { isLoading, registerUser } = useAuth()
+  const [error, setError] = useState<Error>({})
+
+  const handleRegister = async (event: any) => {
+    event.preventDefault()
+    if (event.target[1].value.length >= 8) {
+      if (event.target[1].value === event.target[2].value) {
+        setError({})
+        await registerUser(event.target[0].value, event.target[1].value)
+      } else {
+        setError({
+          id: 1,
+          message: "Hasła nie są takie same",
+        })
+      }
+    } else {
+      setError({
+        id: 2,
+        message: "Hasło musi mieć 8 znaków",
+      })
+    }
+  }
+  if (isLoading) return <></>
```

Comment:
isLoading bym usunął i dodał to wcześniej (np. w router.tsx, aby zwracało pustą stronę przed załadowaniem usera)